src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -279,38 +282,51 @@ private static void RecordRequestStartEventLog(HttpContext httpContext)
             {
                 return null;
             }
-
             var headers = httpContext.Request.Headers;
-            var requestId = headers.TraceParent;
-            if (requestId.Count == 0)
-            {
-                requestId = headers.RequestId;
-            }
+            _propagator.ExtractTraceIdAndState(headers,
+                static (object? carrier, string fieldName, out string? fieldValue, out IEnumerable<string>? fieldValues) =>
+                {
+                    fieldValue = default;
+                    fieldValues = default;
+                    if (carrier is null)
```

Comment:
> if (carrier is null) [](http://example.com/codeflow?start=19&length=21)

may be you reverse this check like 

if (carrier is not null)
{
    var headers = (IHeaderDictionary)carrier;
    fieldValue = headers[fieldName];
    return;
}

so you initlaize fieldValue once instead of twice. this apply to baggage cal too.

Reply:
I'll invert the if and add the null-forgiving operating

Reply:
I meant remove the if